src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -96,6 +97,55 @@ protected function createReadPdo(array $config)
      */
     protected function createPdoResolver(array $config)
     {
+        if (array_key_exists('host', $config)) {
+            return $this->createPdoResolverWithHosts($config);
+        }
+
+        return $this->createPdoResolverWithoutHosts($config);
+    }
+
+    /**
+     * Create a new Closure that resolves to a PDO instance with a specific host or an array of hosts.
+     *
+     * @param  array  $config
+     * @return \Closure
+     */
+    protected function createPdoResolverWithHosts(array $config)
+    {
+        return function () use ($config) {
+            if (! is_array($config['host'])) {
+                $hosts = [$config['host']];
+            } else {
+                $hosts = $config['host'];
+                shuffle($hosts);
+            }
+
+            foreach ($hosts as $host) {
+                $config['host'] = $host;
+
+                try {
+                    return $this->createConnector($config)->connect($config);
+                } catch (PDOException $e) {
```

Comment:
I guess it could suppress other exceptions.
Letting the exception bubble up would kind of defeat the purpose of having a failover.
Would logging connection errors be enough, or do we need to actually show the user why a specific host could not be connected to?